### PR TITLE
Specify use of 400 errors by ASs and 401 errors by RSs

### LINKED ISSUE #93: invalid_dpop_proof and use_dpop_nonce http status response codes
These two new error codes are defined for both token error response and resource error response.

I believe when returned by the token endpoint the intended HTTP Status code is 400.

I believe this is likewise the intention for resource error response, correct? 

~If that's the case then `Figure 15` example should be updated for two reasons~

~1) to actually use `invalid_dpop_proof`, it now uses `invalid_token` (#94)~
~2) to be a 400, not a 401~

Likewise `Figure 20` to use 400, not 401.

If it is intended for token responses to be 400 but resource error responses to be 401 that's okay too, nevertheless going through an implementation in an OIDC server this subject confused me since this server has both the token endpoint as well as a protected resource (userinfo endpoint) and so they re-use errors and their status codes.

### PR DESCRIPTION
**Pull request by selfissued**
Fixes #93

### CODE REVIEW

File: main.md [hunk 1]
```diff
@@ -940,7 +940,7 @@ Of course, each time this happens it requires an extra protocol round trip.
 
 A more efficient manner of supplying a new nonce value is also defined --
 by including a `DPoP-Nonce` HTTP header
-in the `200 OK` response from the previous request.
+in the HTTP `200 (OK)` response from the previous request.
```
Comment (danielfett) [inline main.md:943]: I would suggest to only have the error code, but not the description, within the backticks.

File: main.md [hunk 2]
```diff
@@ -966,6 +966,9 @@ Resource servers can also choose to provide a nonce value to be included
 in DPoP proofs sent to them.
 They provide the nonce using the `DPoP-Nonce` header in same way that authorization servers do.
 The error signaling is performed as described in (#http-auth-scheme).
+Resource servers use an HTTP `401 (Unauthorized)` error code
+with an accompanying `WWW-Authenticate: PoP` value
```
Comment (danielfett) [inline main.md:970]:
```suggestion
with an accompanying `WWW-Authenticate: DPoP` value
```

**Review by danielfett — CHANGES_REQUESTED** [review]

File: main.md (hunk 1 above)
Comment (bc-pi) [inline main.md:943]:
```suggestion
in the HTTP `200` (OK) response from the previous request.
```

Comment (bc-pi) [reply main.md:943]: Agree.  Hopefully help avoid someone thinking the status message itself is a normative requirement. And yes, I've seen that happen.

**Review by b---c — APPROVED** [review]

**Pull request merged**